### ISSUE TITLE
mp3 files with a side data stream can't be added

### ISSUE THREAD

**Issue by leerspace** [icidasset/ongaku-ryoho#7]
While trying to figure out why some of my audio tracks weren't being imported into the OR track list, I noticed that the mp3 files that weren't be imported were the ones with an embedded side data stream (like album art). These mp3s play fine in the browser if accessed directly, but they seem to be getting filtered out while processing the sources.

I can provide an example if needed, but I hesitate to put a hash pointing to a copyrighted song in this GitHub issue.

Here's an example of a file that **cannot** be added (as seen by `ffprobe`):
```
Input #0, mp3, from '04 Blankets of Night.mp3':
  ...
  Duration: 00:07:59.74, start: 0.025056, bitrate: 259 kb/s
    Stream #0:0: Audio: mp3, 44100 Hz, stereo, s16p, 255 kb/s
    Metadata:
      encoder         : LAME3.97 
    Side data:
      replaygain: track gain - -6.900000, track peak - unknown, album gain - unknown, album peak - unknown, 
    Stream #0:1: Video: mjpeg, yuvj420p(pc, bt470bg/unknown/unknown), 600x600 [SAR 1:1 DAR 1:1], 90k tbr, 90k tbn, 90k tbc
    Metadata:
      comment         : Cover (front)
```

Here's an example of a file that **can** be added:
```
Input #0, mp3, from '01 Before the Storm.mp3':
  ...
  Duration: 00:01:09.09, start: 0.025057, bitrate: 320 kb/s
    Stream #0:0: Audio: mp3, 44100 Hz, stereo, s16p, 320 kb/s
    Metadata:
      encoder         : LAME3.93 
```

**Comment by icidasset** [comment]
Hey John. This should work though, were you using IPFS and my fork? You can always send me the file via icid.asset@gmail.com.

**Comment by icidasset** [comment]
<img width="1440" alt="screen shot 2017-07-23 at 12 33 24 am" src="https://user-images.githubusercontent.com/296665/28494962-a44eaea4-6f3e-11e7-9be6-069db4e4afc1.png">

I just ran `ipfs get ...` with the hash you sent me and then added the hash in the app, seems to work fine here 🤔  Send me the console logs if you still encounter the issue.

PS. I really hope my PR for ipfs-go gets merged soon 😞

**Comment by leerspace** [comment]
That's interesting. It's still not working for me (I also tried clearing out Local Storage and starting from scratch and doing an `ipfs get ...` beforehand like you), but the errors I'm getting look like the errors you saw in [this post](https://discuss.ipfs.io/t/music-player-with-ipfs-backend/793/14) when not running your fork.

[Here](https://gist.github.com/leerspace/d1b96d6a3b86dece3a32e21c3082aa33) is the console output I get after clicking on "Process sources".

Maybe I did something wrong when [building go-ipfs from your fork](https://discuss.ipfs.io/t/music-player-with-ipfs-backend/793/8). Some of the IPFS hashes with music I've been testing with work fine, so if this is user error on my part (which is starting to seem much more likely) I can just hold out until your PR for go-ipfs gets merged.

**Comment by icidasset** [comment]
Sorry that it's still not working 😞 I wish that we didn't have to do this CORS setup for all the services we want to integrate. Anyhow, I nudged the ipfs-go folks again to review my PR. Hopefully, it'll be merged soon. *Fingers crossed*

And yeah exactly, this is definitely because the `Content-Range` header isn't returned from the IPFS server. Can you copy-paste your IPFS config here? You can get it with the `ipfs config show` command.

Thanks for going through all this trouble 😅

**Comment by leerspace** [comment]
A potential alternative to using the go-ipfs daemon (which would ideally get rid of the CORS issues) would be to use [js-ipfs](https://github.com/ipfs/js-ipfs) to run an ipfs node in the browser itself. I'm not sure this wouldn't have its own set of issues with what's currently available, but from what I can gather it's one of the more active ipfs implementations.

[Here](https://gist.github.com/leerspace/b10430035cd07a0c538de14a8d4ef8b2) is the output of `ipfs config show`.

Thanks for the help! I'm hoping I didn't do something really stupid, but I wouldn't be surprised :).

**Comment by icidasset** [comment]
Thanks, I'll look into that!

Ok cool, looks the same as my config. You could try setting the following as well, but I'm not sure that's going to do anything (in addition to the ones in the OR CORS instructions):

```
ipfs config --json API.HTTPHeaders.Access-Control-Allow-Methods '["GET", "HEAD"]'
ipfs config --json API.HTTPHeaders.Access-Control-Allow-Headers '["X-Requested-With", "Range", "Content-Range"]'
ipfs config --json Gateway.HTTPHeaders.Access-Control-Allow-Methods '["GET", "HEAD"]'
```

Otherwise, I've got no clue 😥 
In any case, I'll let you know when that PR has been merged.

**Comment by icidasset** [comment]
By the way, one way to make sure that this is the actual issue, is to check your network in tab in your browser's dev tools and look for this:

<img width="1153" alt="screen shot 2017-07-23 at 2 26 33 pm" src="https://user-images.githubusercontent.com/296665/28499321-5159eb6c-6fb3-11e7-8458-3dc53749bf83.png">

**Comment by leerspace** [comment]
Success! Though I don't know for sure what I did different :sweat_smile:. 

The main change was that while I was in dev tools I checked the "disable cache" setting in the network tab, but I'm not quite sure how that would account for this change in behavior (besides _something_ maybe being cached incorrectly due to my attempts to add these files previously with the vanilla go-ipfs daemon). I've confirmed I'm seeing the content-range in the response headers -- or at least I am now.

Thanks for the troubleshooting help!
